src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2304,6 +2304,110 @@ class InsertStatementSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class MergeStatementSegment(BaseSegment):
+    """A `MERGE` statement."""
+
+    type = "merge_statement"
+    match_grammar = StartsWith(
+        Sequence("MERGE", "INTO"),
+    )
+    parse_grammar = Sequence(
```

Comment:
Originally thought this adds no value as `match_grammar` is same as `parse_grammar`, but upon further digging I see it is needed to be separated. Let's make that clear.

```suggestion
    # Note separate `match_grammar` as overridden in other dialects.
    match_grammar = StartsWith(
        Sequence("MERGE", "INTO"),
    )
    parse_grammar = Sequence(
```

Reply:
done